### PR TITLE
i-bem-dom: implement detach() method

### LINKED ISSUE #1102: i-bem__dom: implement `detach` method

### PR DESCRIPTION
**Pull request by narqo**
port #1103 into v3
close #1102

### CODE REVIEW

File: common.blocks/i-bem-dom/i-bem-dom.js [hunk 1]
```diff
@@ -905,9 +914,8 @@ bemDom = /** @exports */{
         });
 
         // NOTE: it was moved here as jquery events aren't triggered on detached DOM elements
-        excludeSelf?
-            ctx.empty() :
-            ctx.remove();
+        _destructDom &&
+            (excludeSelf? ctx.empty() : ctx.remove());
```
Comment (narqo) [inline common.blocks/i-bem-dom/i-bem-dom.js:910]: @dfilatov do you remember why can't we do similar things in `v2` in the same way?

File: common.blocks/i-bem-dom/i-bem-dom.js [hunk 2]
```diff
@@ -883,11 +883,20 @@ bemDom = /** @exports */{
...
     destruct : function(ctx, excludeSelf) {
+        this.detach(ctx, excludeSelf, true);
+    },
+
+    /**
+     * Detaches blocks on a fragment of the DOM tree
+     * @param {jQuery} ctx Root DOM node
+     * @param {Boolean} [excludeSelf=false] Exclude the main domElem
+     */
+    detach : function(ctx, excludeSelf, _destructDom) {
```
Comment (dfilatov) [inline common.blocks/i-bem-dom/i-bem-dom.js]: I would prefer not to expose `_destructDom`. Why not as in v2?

File: common.blocks/i-bem-dom/i-bem-dom.js (hunk 1 above)
Comment (dfilatov) [inline common.blocks/i-bem-dom/i-bem-dom.js:910]: It was related to a new event system. But I can't remember how exactly.

File: common.blocks/i-bem-dom/i-bem-dom.js (hunk 2 above)
Comment (narqo) [inline common.blocks/i-bem-dom/i-bem-dom.js]: I thought it would be easier, but guess it's not. Remade

**Comment by dfilatov** [comment]
:ok:

**Pull request merged**